File: 1주차/beomjookim/merge_strings_alternately.py
class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:
        i, j = 0, 0
        res = ""

        while i < len(word1) or i < len(word2):
            res += word1[i] if i < len(word1) else ""
            res += word2[j] if j < len(word2) else ""
            i += 1
            j += 1
        
        return res

File: 1주차/mijip0320/merge_strings_alternately.js
var mergeAlternately = function(word1, word2) {
    if (word1 !== '' && word2 !== '') {
        let result = "";
        for (let i = 0; i < word1.length; i++){

            //merge first letter
            result += word1[i];
            //if there is letter to merge from word2, merge
            if(word2[i])result += word2[i];
        }

        //if word2 is bigger than word1, merge the leftovers
        if (word2.length > word1.length) {
            let leftovers = word2.substring(word1.length, word2.length);
            result += leftovers;
        }  
        console.log(result)
        
    }
};